kfp/kfp_support_lib/src/kfp_support/workflow_support/utils/remote_jobs_utils.py

```diff
@@ -652,101 +407,132 @@ def follow_execution(
-        if n_actors < 1:
-            logger.warning(
-                f"Not enough cpu/gpu/memory to run transform, "
-                f"required cpu {a_options.get('num_cpus', .5)}, available {cluster_cpu}, "
-                f"required memory {a_options.get('memory', 1)}, available {cluster_mem}, "
-                f"required cpu {actor_gpu}, available {cluster_gpu}"
+    # get current namespace
+    ns = KFPUtils.get_namespace()
+    if ns == "":
+        print(f"Failed to get namespace")
+        sys.exit(1)
+    # submit job
+    remote_jobs = RayRemoteJobs(
+        server_url=server_url,
+        http_retries=additional_params.get("http_retries", 5),
+        wait_interval=additional_params.get("wait_interval", 2),
+    )
+    status, error, submission = remote_jobs.submit_job(
+        name=name, namespace=ns, request=e_params, executor=exec_script_name
+    )
+    if status != 200:
+        print(f"Failed to submit job - status: {status}, error: {error}")
+        exit(1)
+
+    print(f"submitted job successfully, submission id {submission}")
+    # print execution log
+    remote_jobs.follow_execution(
+        name=name,
+        namespace=ns,
+        submission_id=submission,
+        data_access=d_access,
+        print_timeout=additional_params.get("wait_print_tmout", 120),
+        job_ready_timeout=additional_params.get("wait_job_ready_tmout", 600),
+    )
+
+
+def execute_ray_jobs(
+    name: str,  # name of Ray cluster
+    d_access: DataAccess,  # data access
+    additional_params: dict[str, Any],
+    e_params: dict[str, Any],
+    exec_script_name: str,
+    server_url: str,
+) -> None:
+    """
+    Execute Ray jobs on a cluster periodically printing execution log. Completes when all Ray job complete.
+    All of the jobs will be executed, although some of the jobs may fail.
+    :param name: cluster name
+    :param d_access: data access class
+    :param additional_params: additional parameters for the job
+    :param e_params: job execution parameters (specific for a specific transform,
+                        generated by the transform workflow)
+    :param exec_script_name: script to run (has to be present in the image)
+    :param server_url: API server url
+    :return: None
+    """
+    # get current namespace
+    ns = KFPUtils.get_namespace()
+    if ns == "":
+        print(f"Failed to get namespace")
+        sys.exit(1)
+    # Create Ray job submitter
+    remote_jobs = RayRemoteJobs(
+        server_url=server_url,
+        http_retries=additional_params.get("http_retries", 5),
+        wait_interval=additional_params.get("wait_interval", 2),
+    )
+    # find config parameter
+    config = None
+    for key in e_params.keys():
+        if key.startswith("data") and key.endswith("config"):
+            config = key
+            break
+    if config is None:
+        print("Could no find config parameter")
+        exit(1)
+    config_value = e_params[config]
```

Comment:
if config_value is None, or it's length is 1, we assume that it is a single job execution

Reply:
Done